src/js/tech/middleware.js

```diff
@@ -0,0 +1,79 @@
+import { assign } from '../utils/obj.js';
+
+const middlewares = {};
+
+export function use(type, middleware) {
+  (middlewares[type] = middlewares[type] || []).push(middleware);
+}
+
+export function setSource(setTimeout, src, next) {
+  setTimeout(() => ssh(src, middlewares[src.type], next), 1);
```

Comment:
will we get this far if a `src` does not have a `type`? Should we be checking for this somewhere?